python_modules/dagster/dagster_tests/definitions_tests/decorators_tests/test_asset_decorator_with_check_specs.py

```diff
@@ -929,3 +930,37 @@ def test_additional_deps_with_multi_asset_decorator() -> None:
         def foo(context: AssetExecutionContext, asset3) -> Iterable:
             yield Output(value=None, output_name="asset1")
             yield Output(value=None, output_name="asset2")
+
+
+def test_arbitary_asset_check_specs() -> None:
```

Comment:
not sure if this already is under test in some other way so might not need another test but should make sure we still error if the flag is not set